src/Utility/RetryScenarioTester.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace SilverStripe\BehatExtension\Utility;
+
+use Behat\Behat\Tester\ScenarioTester;
+use Behat\Gherkin\Node\FeatureNode;
+use Behat\Gherkin\Node\ScenarioInterface;
+use Behat\Testwork\Environment\Environment;
+use Behat\Testwork\Tester\Result\TestResult;
+
+/**
+ * This tester will retry a scenario if it fails within the same test run.
+ * Note that this is different from the following retry solutions:
+ * - RetryableCallHandler using the @retry tag, which gives assertion an additional 3 seconds
+ * - RerunController, which uses the --rerun option to run only previously failed scenarios in a subsequent test run
+ */
+class RetryScenarioTester implements ScenarioTester
+{
+    private ScenarioTester $decoratedTester;
+
+    public function __construct(ScenarioTester $decoratedTester)
+    {
+        $this->decoratedTester = $decoratedTester;
+    }
+
+    public function setUp(Environment $env, FeatureNode $feature, ScenarioInterface $scenario, $skip)
+    {
+        return $this->decoratedTester->setUp($env, $feature, $scenario, $skip);
+    }
+
+    public function test(Environment $env, FeatureNode $feature, ScenarioInterface $scenario, $skip)
+    {
+        $result = $this->decoratedTester->test($env, $feature, $scenario, $skip);
+        if (!$skip && !$result->isPassed()) {
```

Comment:
Especially with asset-admin, I find I often have to rerun 2 or more times before it goes green.
It might be a good idea to add (either as a replacement of the `is_ci` option or as a separate one) an option to indicate how many times scenarios should be retried. In CI I recommend we set that value to at least 2, maybe 3. We can then easily tweak that value in a central place as the need arises, without having to edit this code.

That said, if you think that's not worth doing, I'll be okay with merging this as-is.